src/AdvancedADC.cpp

```diff
@@ -212,12 +273,24 @@ int AdvancedADC::begin(uint32_t resolution, uint32_t sample_rate, size_t n_sampl
     hal_dma_enable_dbm(&descr->dma, descr->dmabuf[0]->data(), descr->dmabuf[1]->data());
     HAL_NVIC_EnableIRQ(descr->dma_irqn);
 
+    if(do_start)
+    {
+        return(start(sample_rate));
+    }
+
```

Comment:
```suggestion
    if (do_start) {
        return start(sample_rate);
    }
```

No `return()` please.